### PR TITLE
Paddle_OP_Not_euqal

### PR DESCRIPTION
**Pull request by OpenVINO-dev-contest**
### Details:
 - Add new Paddle operation: [Not_euqal](https://www.paddlepaddle.org.cn/documentation/docs/en/1.8/api/layers/not_equal.html)

### Tickets:
 - *ticket-id*

Screenshot from 2022-07-03 19-47-54

### CODE REVIEW

File: src/frontends/paddle/src/op/elementwise_ops.cpp [hunk 1]
```diff
@@ -46,6 +46,10 @@ NamedOutputs elementwise_greater_equal(const NodeContext& node_context) {
     return elementwise_ops<default_opset::GreaterEqual>(node_context);
 }
 
+NamedOutputs elementwise_not_equal(const NodeContext& node_context) {
+    return elementwise_ops<default_opset::NotEqual>(node_context);
```
Comment (liubo-intel) [inline src/frontends/paddle/src/op/elementwise_ops.cpp:50]: “NotEqual” Operator can reuse 'elementwise_ops' base process, but such kind of elementwise related Ops are recommanded to use separate conversions(like, [greater_than](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/paddle/src/op/greater_than.cpp) )

File: src/core/tests/frontend/paddle/test_models/gen_scripts/generate_not_equal.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    saveModel(name, exe, feedkeys=['x', 'y'], fetchlist=[out],
+              inputs=[x, y], outputs=[outs[0]], target_dir=sys.argv[1])
+
+    return outs[0]
+
+
+def main():
+    import paddle as paddle
+    data_x = np.array([[[[-1, 0, 1]], [[2, 3, 4]]]]).astype(np.float32)
+    data_y = np.array([[[[2, 0, 3]], [[3, 1, 4]]]]).astype(np.float32)
```
Comment (liubo-intel) [inline src/core/tests/frontend/paddle/test_models/gen_scripts/generate_not_equal.py:39]: looks like this Paddle Op supports 'float32，float64，int32，int64' data type, so except 'float64' is limited by current test framework, we would better cover its supported data type as much as possible(like ['greater_than' test case](https://github.com/openvinotoolkit/openvino/blob/d22c429d0e567b80f1c5cbcde50b830fbbfac029/src/core/tests/frontend/paddle/test_models/gen_scripts/generate_greater_than.py#L43))

**Review by liubo-intel — COMMENTED** [review]
Hi, @OpenVINO-dev-contest : I think it will be helpful for reviewers to conveniently learn about the Paddle Op's spec if we add its spec link(for some paddle legacy op which we can't find its 2.1 version spec, we can use its legacy version spec or its source code link) in PR's 'Details:' description part. e.g.
not_equal: https://www.paddlepaddle.org.cn/documentation/docs/en/1.8/api/layers/not_equal.html

File: src/core/tests/frontend/paddle/test_models/gen_scripts/generate_not_equal.py (hunk 2 above)
Comment (OpenVINO-dev-contest) [reply src/core/tests/frontend/paddle/test_models/gen_scripts/generate_not_equal.py]: updated accordingly

**Review by liubo-intel — APPROVED** [review]

File: src/frontends/paddle/src/op/elementwise_ops.cpp (hunk 1 above)
Comment (meiyang-intel) [inline src/frontends/paddle/src/op/elementwise_ops.cpp:50]: 1. OpenVINO NotEqual op return the type the same as input type while Paddle not_equal op return the bool type. The return type is different. Will it really work in a paddle model containing 'not_equal'?
2. paddle 'not_equal' has an optional input 'cond'. How to deal with the 'not_equal' with 'cond' input?

Comment (OpenVINO-dev-contest) [reply src/frontends/paddle/src/op/elementwise_ops.cpp:50]: Hi, we use this OP spec as its reference implementation
https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/not_equal_en.html#not-equal

Comment (meiyang-intel) [reply src/frontends/paddle/src/op/elementwise_ops.cpp:50]: Got it. What about the question 1?

Comment (OpenVINO-dev-contest) [reply src/frontends/paddle/src/op/elementwise_ops.cpp:50]: This OP's return format is same as other elewise OPs, (e.g Equal), if Equal works, it should work as well.
https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/equal_en.html

Comment (OpenVINO-dev-contest) [reply src/frontends/paddle/src/op/elementwise_ops.cpp:50]: @meiyang-intel any suggestion?

**Review by ceciliapeng2011 — APPROVED** [review]

**Review by meiyang-intel — APPROVED** [review]

**Review by ceciliapeng2011 — APPROVED** [review]

File: src/frontends/paddle/src/op/elementwise_ops.cpp [hunk 3]
```diff
@@ -46,6 +46,11 @@ NamedOutputs elementwise_greater_equal(const NodeContext& node_context) {
     return elementwise_ops<default_opset::GreaterEqual>(node_context);
 }
 
+
```
Comment (meiyang-intel) [inline src/frontends/paddle/src/op/elementwise_ops.cpp:49]:
```suggestion
```

**Review by meiyang-intel — CHANGES_REQUESTED** [review]

**Review by meiyang-intel — APPROVED** [review]

**Comment by meiyang-intel** [comment]
Hi @OpenVINO-dev-contest please resolve the conflicts

**Comment by ceciliapeng2011** [comment]
@OpenVINO-dev-contest please check CI failures -
[ RUN      ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_float32
../../openvino/src/frontends/tests/frontend/shared/src/op_fuzzy.cpp:119: Failure
Expected: doLoadFromFile() doesn't throw an exception.
  Actual: it throws.
[  FAILED  ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_float32, where GetParam() = ("paddle", "test_model_zoo/paddle_test_models/", "not_equal_float32") (1 ms)
[ RUN      ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_int32
../../openvino/src/frontends/tests/frontend/shared/src/op_fuzzy.cpp:119: Failure
Expected: doLoadFromFile() doesn't throw an exception.
  Actual: it throws.
[  FAILED  ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_int32, where GetParam() = ("paddle", "test_model_zoo/paddle_test_models/", "not_equal_int32") (0 ms)
[ RUN      ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_int64
../../openvino/src/frontends/tests/frontend/shared/src/op_fuzzy.cpp:119: Failure
Expected: doLoadFromFile() doesn't throw an exception.
  Actual: it throws.
[  FAILED  ] PaddleFuzzyOpTest/FrontEndFuzzyOpTest.testOpFuzzy/paddle_not_equal_int64, where GetParam() = ("paddle", "test_model_zoo/paddle_test_models/", "not_equal_int64") (0 ms)

**Review by ilyachur — APPROVED** [review]

**Pull request merged**